### ISSUE TITLE
InjectManifest/swSrc Bug

### ISSUE THREAD

**Issue by lolcoolkat** [hanford/next-offline#175]
Found a bug after setting `generateSW` to false and `workboxOpts.swSrc`. If you set generateSW to false.. it will end up using `InjectManifest` instead of `GenerateSW`; however, `InjectManifest` uses the same default `defaultWorkboxOpts` that `generateSW` uses. `InjectManifest` doesn't recognize `runtimeCaching` or `inlineWorkboxRuntime` so my build crashes.

GenerateSW default config (This one works fine)
```
const defaultWorkboxOpts = {
  exclude: preCacheManifestBlacklist,
  // As of Workbox v5 Alpha there isn't a well documented way to move workbox runtime into the directory
  // required by Next. As a work around, we inline the tree-shaken runtime into the main Service Worker file
  // at the cost of less cacheability
  inlineWorkboxRuntime: true,
  modifyURLPrefix: {
    'static/': nextAssetLinkPrefix,
  },
  runtimeCaching: [
    {
      urlPattern: /^https?.*/,
      handler: 'NetworkFirst',
      options: {
        cacheName: 'offlineCache',
        expiration: {
          maxEntries: 200
        }
      }
    }
  ],
};
```

InjectManifest Default Config
```

const defaultWorkboxOpts = {
  exclude: preCacheManifestBlacklist,
  // As of Workbox v5 Alpha there isn't a well documented way to move workbox runtime into the directory
  // required by Next. As a work around, we inline the tree-shaken runtime into the main Service Worker file
  // at the cost of less cacheability
  inlineWorkboxRuntime: true, <-- Can't use this in InjectManifest
  modifyURLPrefix: {
    'static/': nextAssetLinkPrefix,
  },
  runtimeCaching: [ <-- Can't use this in InjectManifest
    {
      urlPattern: /^https?.*/,
      handler: 'NetworkFirst',
      options: {
        cacheName: 'offlineCache',
        expiration: {
          maxEntries: 200
        }
      }
    }
  ],
};
```

If I take out `runtimeCaching` and `inlineWorkboxRuntime`, it's able to build with a `swSrc` file I specify.

**Comment by lolcoolkat** [comment]
tl;dr - I can't use `swSrc` with `generateSW`

It gives me an error saying `runtimeCaching` and `inlineWorkboxRuntime` are invalid parameters. However I don't have them set in my `workboxOpts`

**Comment by hanford** [comment]
@lolcoolkat what version of next-offline are you using?

**Comment by lolcoolkat** [comment]
@hanford was using ^5.0.0-alpha.0

**Comment by lolcoolkat** [comment]
@hanford btw I tried 5.0.0-beta.2 and still the same thing..

here's what I ended up doing to fix it.. and it works fine now

```
    // If using InjectManifest, remove the following options
    if (!generateSw) {
      delete defaultWorkboxOpts.inlineWorkboxRuntime
      delete defaultWorkboxOpts.runtimeCaching
    }

    const combinedWorkboxOpts = { ...defaultWorkboxOpts, ...workboxOpts }
    const skipDuringDevelopment = options.dev && !generateInDevMode
```

**Comment by hanford** [comment]
This should be fixed in `next-offline@5.0.0-beta.3`

Thanks for reporting @lolcoolkat
